core/io/image.cpp

```diff
@@ -2719,6 +2718,33 @@ void Image::fill(const Color &c) {
 	}
 }
 
+void Image::fill_rect(const Rect2 &p_rect, const Color &p_color) {
+	ERR_FAIL_COND_MSG(!_can_modify(format), "Cannot fill rect in compressed or custom image formats.");
+
+	Rect2i r = Rect2i(0, 0, width, height).intersection(p_rect.abs());
+	if (r.has_no_area()) {
+		return;
+	}
+
+	uint8_t *dst_data_ptr = data.ptrw();
+
+	int pixel_size = get_format_pixel_size(format);
+
+	// Put first pixel with the format-aware API.
+	set_pixelv(r.position, p_color);
+	const uint8_t *rect_first_pixel_ptr = &dst_data_ptr[(r.position.y * width + r.position.x) * pixel_size];
+
+	for (int y = r.position.y; y < r.position.y + r.size.y; y++) {
+		for (int x = r.position.x; x < r.position.x + r.size.x; x++) {
+			uint8_t *dst = &dst_data_ptr[(y * width + x) * pixel_size];
+
+			for (int k = 0; k < pixel_size; k++) {
+				dst[k] = rect_first_pixel_ptr[k];
+			}
```

Comment:
I believe this could just be
```suggestion
			memcpy(dst, rect_first_pixel_ptr, pixel_size);
```

Reply:
Makes perfect sense. I've kinda just copy-pasted `Image::fill()` and modified it accordingly. It uses loop for copying too:
https://github.com/godotengine/godot/blob/905567a9bda93bf9482debbcc644ac32eed7d087/core/io/image.cpp#L2714-L2716
I wonder if there's some reason behind it. :thinking: I don't see one. So I'm all for changing both to use `memcpy`. I'll wait for a core member's opinion on this though.